.github/workflows/kryoptic-fips.yml

```diff
@@ -0,0 +1,137 @@
+          key: fedora-dnf-${{ steps.get-date.outputs.date }}
+
+      #####################
+      ### OpenSSL build ###
+      #####################
+      - name: Setup OpenSSL 3.5
+        id: ossl-setup
+        run: |
+          OPENSSL_BRANCH="openssl-3.5"
+
+          cd ..
+          git clone https://github.com/openssl/openssl.git \
+                    --single-branch --branch $OPENSSL_BRANCH openssl
+          cd openssl
+          echo "KRYOPTIC_OPENSSL_SOURCES=$PWD" >> "$GITHUB_ENV"
+          echo "cacheid=${{ runner.os }}-ossl-$(git rev-parse HEAD)" >> "$GITHUB_OUTPUT"
+
+      - name: Restore OpenSSL build if cached
+        uses: actions/cache/restore@v4
+        id: cache
+        with:
+          path: ${{ env.KRYOPTIC_OPENSSL_SOURCES }}
+          key: ${{ steps.ossl-setup.outputs.cacheid }}
+
+      - name: Build OpenSSL
+        if: ${{ steps.cache.outputs.cache-hit != 'true' }}
+        run: |
+          pushd ${{ env.KRYOPTIC_OPENSSL_SOURCES }}
+          ./Configure
+          make
+
+      - name: Cache OpenSSL 3.5 build
+        if: ${{ steps.cache.outputs.cache-hit != 'true' }}
+        uses: actions/cache/save@v4
+        with:
+          path: ${{ env.KRYOPTIC_OPENSSL_SOURCES }}
+          key: ${{ steps.ossl-setup.outputs.cacheid }}
+
+      ######################
+      ### kryoptic build ###
+      ######################
+      - name: Setup kryoptic
+        run: |
+          KRYOPTIC_REVISION="v1.3.1"
+
+          cd ..
+          git clone https://github.com/latchset/kryoptic.git \
+              --depth 1 --single-branch --revision $KRYOPTIC_REVISION kryoptic
+
+      - name: Generate lock file
+        run: |
+          cd ../kryoptic &&
+          cargo generate-lockfile
+
+      - name: Cache Rust dependencies
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            ../kryoptic/target/
+          key: fedora-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Build kryoptic
+        run: |
+          FEATURES="fips,pqc,dummy-integrity"
+          OPTS="--no-default-features"
+
+          cd ../kryoptic &&
+          cargo build -vv $OPTS --features "$FEATURES"
+
+      - uses: actions/upload-artifact@v4
+        if: failure()
+        with:
+          name: Build logs OpenSSL version 3.5
+          path: |
+            target/debug/build/*/output
+
+      - name: Checkout rust-cryptoki
+        uses: actions/checkout@v4
+
+      #################
+      ### the tests ###
+      #################
+      - name: Run test script
+        env:
+          KRYOPTIC_CONF: /tmp/kryoptic.sql
+          TEST_PKCS11_MODULE: /__w/rust-cryptoki/kryoptic/target/debug/libkryoptic_pkcs11.so
+          OUT_DIR: /__w/rust-cryptoki/kryoptic/target/debug/deps/
+        run: |
+          RUST_BACKTRACE=1 cargo build --all-features &&
```

Comment:
I think `RUST_BACKTRACE=1` can also be moved to `env`.
I don't think we need `cargo build` here... `cargo check` should be sufficient :thinking: